### PR TITLE
OpenX: accept incoming string fields to support Prebid.js 9 

### PR DESCRIPTION
**Pull request by laurb9**
1. accept incoming `imp[].ext.customFloor` as string or float, emit float in `imp[].bidfloor`
2. accept incoming `imp[].ext.unit` as string or integer, emit string in `imp[].tagid`
3. enable gzip compression

1,2 are to support the Prebid.js 9 changes introduced in https://github.com/prebid/Prebid.js/pull/11458

Corresponding PBS-Java change: https://github.com/prebid/prebid-server-java/pull/3178

### CODE REVIEW

File: adapters/openx/openxtest/supplemental/multi-imp.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
         "banner": {},
         "ext": {
           "bidder": {
-            "unit": "banner-unit-1",
```
Comment (SyntaxNode) [inline adapters/openx/openxtest/supplemental/multi-imp.json:10]: Was this test case previously invalid? Doesn't look like it adhered to the unit json schema pattern.

File: adapters/openx/openxtest/supplemental/multi-imp.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
+            "unit": "111",
```
Comment (SyntaxNode) [inline adapters/openx/openxtest/supplemental/multi-imp.json:10]: Curious, why "111" instead of "1"? Is there a test involved with the length of the unit string?

File: adapters/openx/params_test.go [hunk 3]
```diff
@@ -56,7 +59,8 @@ var invalidParams = []string{
 	`{"unit": "123", "delDomain": "foo.b"}`,
 	`{"unit": "123", "delDomain": "foo.barr"}`,
 	`{"unit": "123", "delDomain": ".bar"}`,
-	`{"unit": "123", "delDomain": "foo.bar", "customFloor": "0.1"}`,
+	`{"unit": "123", "delDomain": "foo.bar", "customFloor": ""}`,
+	`{"unit": "123", "delDomain": "foo.bar", "customFloor": "-0.1"}`,
 	`{"unit": "123", "delDomain": "foo.bar", "customFloor": -0.1}`,
 	`{"unit": "123", "delDomain": "foo.bar", "customParams": "foo: bar"}`,
```
Comment (SyntaxNode) [inline adapters/openx/params_test.go:65]: I recommend adding customFloor tests invalid test cases as "1." and "1.0x" to add more test coverage of the regex.

File: adapters/openx/openxtest/supplemental/multi-imp.json (hunk 2 above)
Comment (laurb9) [reply adapters/openx/openxtest/supplemental/multi-imp.json:10]: No functional reason, just think it's easier to find and replace if needed.

File: adapters/openx/openxtest/supplemental/multi-imp.json (hunk 1 above)
Comment (laurb9) [reply adapters/openx/openxtest/supplemental/multi-imp.json:10]: I don't recall if these were valid at some point in the past, but they are not now.

**Comment by laurb9** [comment]
Please let me know if there is anything else we should do

File: endpoints/openrtb2/sample-requests/disabled/good/partial.json [hunk 4]
```diff
@@ -43,7 +43,7 @@
           "openx": {
             "unit": "539439964",
             "delDomain": "se-demo-d.openx.net",
-            "customFloor": 0.1,
+            "customFloor": "0.5",
             "customParams": {"foo": "bar"}
           }              
         }
```
Comment (onkarvhanumante) [inline endpoints/openrtb2/sample-requests/disabled/good/partial.json:49]: @laurb9 why was this file changed? Adapter change should be limited to Adapter package

File: static/bidder-params/openx.json [hunk 5]
```diff
@@ -22,9 +23,10 @@
       "format": "uuid"
     },
     "customFloor": {
-      "type": "number",
+      "type": ["number", "string"],
       "description": "The minimum CPM price in USD.",
-      "minimum": 0
```
Comment (onkarvhanumante) [inline static/bidder-params/openx.json:27]: should add json tests 
- to have request where unit is string 
- to have request where unit is number 
- to have request where customFloor is string 
- - to have request where customFloor is number

File: endpoints/openrtb2/sample-requests/disabled/good/partial.json (hunk 4 above)
Comment (laurb9) [reply endpoints/openrtb2/sample-requests/disabled/good/partial.json:49]: This file contains an OpenX adapter request, and we thought it would be best if it were updated to the most recent request format.

File: static/bidder-params/openx.json (hunk 5 above)
Comment (laurb9) [reply static/bidder-params/openx.json:27]: These parameter variations are in `optional-params.json`:
`unit` is string: https://github.com/prebid/prebid-server/pull/3668/files#diff-4214c2dc95f1c24891a70e424d9576ab40f10447745b95f68a877ca7145f1bd5R12
`unit` is number: https://github.com/prebid/prebid-server/pull/3668/files#diff-4214c2dc95f1c24891a70e424d9576ab40f10447745b95f68a877ca7145f1bd5R28
`customFloor` is string: https://github.com/prebid/prebid-server/pull/3668/files#diff-4214c2dc95f1c24891a70e424d9576ab40f10447745b95f68a877ca7145f1bd5R15
`customFloor` is number: https://github.com/prebid/prebid-server/pull/3668/files#diff-4214c2dc95f1c24891a70e424d9576ab40f10447745b95f68a877ca7145f1bd5R31

File: endpoints/openrtb2/sample-requests/disabled/good/partial.json (hunk 4 above)
Comment (onkarvhanumante) [reply endpoints/openrtb2/sample-requests/disabled/good/partial.json:49]: could move this to openx specific exemplary test folder

**Comment by onkarvhanumante** [comment]
@laurb9 requesting to take a look at https://github.com/prebid/prebid-server/pull/3668/files#r1652722717

File: endpoints/openrtb2/sample-requests/disabled/good/partial.json (hunk 4 above)
Comment (laurb9) [reply endpoints/openrtb2/sample-requests/disabled/good/partial.json:49]: I can undo this particular line change if it blocks the review, but moving it to the openx test folder does not add any new use cases there. Removing samples from `openrtb2/sample-requests` is outside the scope of this PR, aside from that `openrtb2/sample-requests` contains requests for other adapters as well, so the file looks consistent with the intended purpose of that directory, not as an adapter-specific request, but an usage example for certain pbs configurations.

**Review by onkarvhanumante — APPROVED** [review]

File: openrtb_ext/imp_openx.go [hunk 6]
```diff
@@ -1,10 +1,12 @@
...
+import "encoding/json"
+
 // ExtImpOpenx defines the contract for bidrequest.imp[i].ext.prebid.bidder.openx
 type ExtImpOpenx struct {
-	Unit         string                 `json:"unit"`
+	Unit         json.Number            `json:"unit"`
 	Platform     string                 `json:"platform"`
 	DelDomain    string                 `json:"delDomain"`
-	CustomFloor  float64                `json:"customFloor"`
+	CustomFloor  json.Number            `json:"customFloor"`
```
Comment (onkarvhanumante) [inline openrtb_ext/imp_openx.go:10]: @laurb9 
could use `StringInt` data type for Unit, CustomFloor
https://github.com/prebid/prebid-server/blob/a1f48b54af8e831927a0f6838f56493fa03efb08/util/jsonutil/stringInt.go#L7-L29

**Review by onkarvhanumante — CHANGES_REQUESTED** [review]
@laurb9  ptal at  https://github.com/prebid/prebid-server/pull/3668/files#r1660697479

**Review by onkarvhanumante — APPROVED** [review]

**Comment by laurb9** [comment]
Just checking if there is anything else to be done here, this PR addresses breaking changes in Prebid.js 9

**Review by Sonali-More-Xandr — APPROVED** [review]
